### PR TITLE
Use lsp-protocol and plists

### PR DESCRIPTION
**Pull request by ericdallo**
https://github.com/emacs-lsp/lsp-mode/issues/1751

Depends on https://github.com/emacs-lsp/lsp-mode/pull/1747

### CODE REVIEW

File: lsp-ui-sideline.el [hunk 1]
```diff
@@ -222,18 +223,18 @@ function signature)."
     (cond
      ((stringp contents) contents)
      ((sequencep contents) ;; MarkedString[]
```
Comment (yyoncho) [inline lsp-ui-sideline.el:225]: sequencep will match plists. I think it should be changed to vectorp

File: lsp-ui-doc.el [hunk 2]
```diff
@@ -294,12 +297,12 @@ We don't extract the string that `lps-line' is already displaying."
                ;; (propertize "\n\n" 'face '(:height 0.4))
                ))
    ;; when we get markdown contents, render using emacs gfm-view-mode / markdown-mode
-   ((string= (gethash "kind" contents) "markdown") ;; Markdown MarkupContent
-    (lsp-ui-doc--extract-marked-string (gethash "value" contents) "markdown"))
-   ((gethash "kind" contents) (gethash "value" contents)) ;; Plaintext MarkupContent
-   ((gethash "language" contents) ;; MarkedString
-    (lsp-ui-doc--extract-marked-string (gethash "value" contents)
-                                       (gethash "language" contents)))))
+   ((string= (lsp:markup-content-kind contents) "markdown") ;; Markdown MarkupContent
```
Comment (kiennq) [inline lsp-ui-doc.el:300]:
```suggestion
   ((string= (lsp:markup-content-kind contents) lsp/markup-kind-markdown) ;; Markdown MarkupContent
```

File: lsp-ui-sideline.el [hunk 3]
```diff
@@ -221,19 +222,19 @@ function signature)."
...
-     ((sequencep contents) ;; MarkedString[]
-      (seq-find (lambda (it) (and (hash-table-p it)
-                                  (lsp-get-renderer (gethash "language" it))))
+     ((vectorp contents) ;; MarkedString[]
+      (seq-find (lambda (it) (and (lsp-marked-string? it)
+                                  (lsp-get-renderer (lsp:marked-string-language it))))
                 contents))
-     ((gethash "kind" contents) contents) ;; MarkupContent
-     ((gethash "language" contents) contents))))
+     ((lsp:markup-content-kind contents) contents) ;; MarkupContent
```
Comment (kiennq) [inline lsp-ui-sideline.el:229]: Shouldn't it better to check the type instead?
```suggestion
     ((lsp-markup-content? contents) contents) ;; MarkupContent
```

Comment (yyoncho) [reply lsp-ui-sideline.el:229]: we do it this way in lsp-mode

File: lsp-ui-doc.el [hunk 4]
```diff
@@ -287,19 +290,19 @@ MarkedString | MarkedString[] | MarkupContent (as defined in the LSP).
...
    ;; when we get markdown contents, render using emacs gfm-view-mode / markdown-mode
-   ((string= (gethash "kind" contents) "markdown") ;; Markdown MarkupContent
-    (lsp-ui-doc--extract-marked-string (gethash "value" contents) "markdown"))
-   ((gethash "kind" contents) (gethash "value" contents)) ;; Plaintext MarkupContent
-   ((gethash "language" contents) ;; MarkedString
-    (lsp-ui-doc--extract-marked-string (gethash "value" contents)
-                                       (gethash "language" contents)))))
+   ((string= (lsp-markup-content? contents) lsp/markup-kind-markdown) ;; Markdown MarkupContent
+    (lsp-ui-doc--extract-marked-string (lsp:markup-content-value contents) lsp/markup-kind-markdown))
+   ((lsp:markup-content-kind contents) (lsp:markup-content-value contents)) ;; Plaintext MarkupContent
```
Comment (kiennq) [inline lsp-ui-doc.el:302]:
```suggestion
   ((lsp-markup-content? contents) (lsp:markup-content-value contents)) ;; Plaintext MarkupContent
```

File: lsp-ui-sideline.el [hunk 5]
```diff
@@ -221,19 +222,19 @@ function signature)."
...
-      (seq-find (lambda (it) (and (hash-table-p it)
-                                  (lsp-get-renderer (gethash "language" it))))
+     ((vectorp contents) ;; MarkedString[]
+      (seq-find (lambda (it) (and (lsp-marked-string? it)
+                                  (lsp-get-renderer (lsp:marked-string-language it))))
                 contents))
-     ((gethash "kind" contents) contents) ;; MarkupContent
-     ((gethash "language" contents) contents))))
+     ((lsp-markup-content? contents) contents) ;; MarkupContent
+     ((lsp:marked-string-language contents) contents))))
```
Comment (kiennq) [inline lsp-ui-sideline.el:230]:
```suggestion
     ((lsp-marked-string? contents) contents))))
```

**Comment by ericdallo** [comment]
Fixed the master byte-compile errors and failing tests, could you take a look later @yyoncho?

**Pull request merged**